### PR TITLE
[2.5.12] Update CSI documentation

### PR DESCRIPTION
**Pull request by doflamingo721**
Updated the CSI documentation with the workaround for upgrading kubernetes version after csi-migration. 
Issue: https://github.com/rancher/rancher/issues/35102

### CODE REVIEW

File: packages/rancher-vsphere-csi/charts/README.md
Comment (PennyScissors) [inline packages/rancher-vsphere-csi/charts/README.md:88]: Can we change this to a paragraph rather than bullet points since it all basically reads like a paragraph? 

This could also be simplified to something like this:

Follow the steps in the [Upgrading Kubernetes](https://rancher.com/docs/rancher/v2.5/en/cluster-admin/upgrading-kubernetes/) guide to upgrade the Kubernetes version after migration.

**Important:** If you have a workload with a bound volume attached to it during an upgrade, you will see the pods get redeployed and the old pods will get stuck in a **Terminating** state. You can safely delete the old stuck pods by executing the command `kubectl delete pod <pod_id> --force=true --timeout=120s --grace-period=0` to fix this problem, but it's imperative you verify before deleting a pod, that there is a new active pod corresponding to the old pod you want to delete. Deleting an old pod before a new one has been deployed could result in loss of data and bound volumes (Due to `--grace-period=0`).

**Review by PennyScissors — COMMENTED** [review]
Validation is failing, but that's expected since you are modifying the chart in place. You will need a committer/admin to force merge this for you once it's ready.

File: packages/rancher-vsphere-csi/package.yaml
Comment (aiyengar2) [inline packages/rancher-vsphere-csi/package.yaml:2]: we should bump this to 1 so we make a change that isn't modifying an already released chart

**Review by aiyengar2 — CHANGES_REQUESTED** [review]

File: packages/rancher-vsphere-csi/charts/README.md
Comment (doflamingo721) [reply packages/rancher-vsphere-csi/charts/README.md]: @PennyScissors I made the changes as you mentioned. Could you please review it again?

File: packages/rancher-vsphere-csi/package.yaml
Comment (doflamingo721) [reply packages/rancher-vsphere-csi/package.yaml]: @aiyengar2 I bumped the packageVerison as you mentioned. Can you please review it again?

File: charts/rancher-vsphere-csi/rancher-vsphere-csi/2.3.1/Chart.yaml
Comment (PennyScissors) [inline charts/rancher-vsphere-csi/rancher-vsphere-csi/2.3.1/Chart.yaml:9]: This should be 2.3.1 to match `version`

Comment (doflamingo721) [reply charts/rancher-vsphere-csi/rancher-vsphere-csi/2.3.1/Chart.yaml:9]: @PennyScissors @aiyengar2  Should update the value to 2.3.1?

Comment (aiyengar2) [reply charts/rancher-vsphere-csi/rancher-vsphere-csi/2.3.1/Chart.yaml:9]: Let's keep it since we've always had it. Let's make it 2.3.1

**Pull request closed without merging**
